### PR TITLE
[Bug] - Deadlock found when trying to get lock on DataObject import

### LINKED ISSUE #15150: [Bug] - Deadlock found when trying to get lock on DataObject import
Hi, 

For your information, we are facing the exact same problem with the `object_meatada` table on one of our object. 

What happens is, we are trying to populate a fresh install of pimcore with a little bit more than 100 000 DataObjects through the DataImporter Bundle. These DataObjects have multiple relation fields. Some of these fields are filled through the DataImporter, but others are filled using an EventListener (using the `PreSaveEvent` of the DataImporter) because what we are trying to achieve is not possible natively with the DataImporter.

When we execute the import in parallel with multiple process (ie. 10), a lot of objects couldn't be created because of this error :
```
Next Doctrine\DBAL\Exception\DeadlockException: An exception occurred while executing 'INSERT INTO object_metadata_OBJECT (o_id, dest_id, fieldname, ownertype, ownername, index, position, type, column, data) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ON DUPLICATE KEY UPDATE o_id = ?, dest_id = ?, fieldname = ?, ownertype = ?, ownername = ?, index = ?, position = ?, type = ?, column = ?, data = ?;' with params [my params]:

SQLSTATE[40001]: Serialization failure: 1213 Deadlock found when trying to get lock; try restarting transaction in /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:51
```

It failed even though it retried, according to the message.

The params (my enveloppes) are good since, to try to debug this, we imported all of our objects by executing the parallel process with 1 process (took 6 hours, but no errors). It is clearly linked to the parallelization of the import.

I also have to mention that we were able to import everything by commenting our event in the DataImporter, with multiple processes (again, 10). So the events in the DataImporter seems to be a part of the problem as well in our case. 

Lastly, the problem doesn't seem to appear when updating already created objects. It only affects the creation.

Should I create a new issue with this problem, or is it also part of this problem ?

_Originally posted by @SamyMP in https://github.com/pimcore/pimcore/issues/15064#issuecomment-1539650108_

### PR DESCRIPTION
**Pull request by robertSt7**
resolves #15150

### CODE REVIEW

**Comment by robertSt7** [comment]
@SamyMP Hi, thank you for reporting https://github.com/pimcore/pimcore/issues/15150. Does this fix it?

**Comment by dvesh3** [comment]
@robertSt7 please rebase on 10.6 branch. thanks!

**Comment by SamyMP** [comment]
Hi @robertSt7, 

I don't know for sure if that solves the issue but from what I have seen, it should, thank you for the PR.
The issue appeared when importing a large amount of data in Pimcore using the data-importer, I'll let you know if I have to do that again after this PR is merged

File: models/DataObject/Data/ObjectMetadata/Dao.php [hunk 1]
```diff
@@ -118,6 +118,7 @@ public function createOrUpdateTable(DataObject\ClassDefinition $class)
         $table = 'object_metadata_' . $classId;
 
         $this->db->executeQuery('CREATE TABLE IF NOT EXISTS `' . $table . "` (
+              `auto_id` int(11) NOT NULL AUTO_INCREMENT,
```
Comment (dvesh3) [inline models/DataObject/Data/ObjectMetadata/Dao.php]:
```suggestion
              `id` int(11) UNSIGNED NOT NULL AUTO_INCREMENT,
```
can we call it just `id`? just to be consistent.

**Comment by dvesh3** [comment]
Hi @SamyMP could you check if this fixes the problem at your end? so that we get second confirmation with different data model. thanks!

**Review by dvesh3 — APPROVED** [review]

File: models/DataObject/Data/ObjectMetadata/Dao.php [hunk 2]
```diff
@@ -118,6 +118,7 @@ public function createOrUpdateTable(DataObject\ClassDefinition $class)
         $table = 'object_metadata_' . $classId;
 
         $this->db->executeQuery('CREATE TABLE IF NOT EXISTS `' . $table . "` (
+              `id` int(11) UNSIGNED NOT NULL AUTO_INCREMENT,
```
Comment (dvesh3) [inline models/DataObject/Data/ObjectMetadata/Dao.php:121]: @robertSt7 this conflicts with changes done on 11.0 https://github.com/pimcore/pimcore/commit/4d93f1f93eb8a5544ea4ae8c4345d5fec181b0f3#diff-6c6cbcb57ed8138d7a968f573cd8bc8a3aeaef3c76bd0ed634a53d2822909712R88

please see if a follow up is required on 11.0. thanks!

**Comment by blankse** [comment]
@robertSt7 @dvesh3 I get errors in the migration after update 10.6.3 to 10.6.4:

```
An exception occurred while executing 'ALTER TABLE object_metadata_dfperson  
   ADD id INT AUTO_INCREMENT NOT NULL, DROP PRIMARY KEY, ADD PRIMARY KEY (id)  
  ':                                                                           
                                                                               
  SQLSTATE[HY000]: General error: 1553 Cannot drop index 'PRIMARY': needed in  
   a foreign key constraint                                    
```

**Comment by dvesh3** [comment]
thanks for reporting. the issue is here https://github.com/pimcore/pimcore/issues/15586

**Pull request merged**